document/src/main/java/com/yahoo/document/idstring/IdString.java

```diff
@@ -42,6 +42,7 @@ public String getGroup() {
     private final String namespace;
     private final String namespaceSpecific;
     private Utf8String cache;
+    static final int MAX_LENGTH_EXCEPT_NAMESPACE_SPECIFIC = 0xfffe;
```

Comment:
`0xffff - 1` due to actual offset being the next char after `:`, I presume? A tiny comment would be nice for future reference 🙂